File: sandbox/fortranpython/navaro/alloc.f90
! from http://calcul.math.cnrs.fr/Documents/Ecoles/2010/f2py-cours.pdf
! Pierre Navaro


module f90module
    implicit none
    real(8), dimension(:), allocatable :: farray
    
contains

    subroutine init( n ) !Allocation du tableau farray
        integer, intent(in) :: n
        allocate(farray(n))
    end subroutine init

    subroutine test_array()
        print*, allocated(farray), size(farray)
    end subroutine test_array
end module f90module

File: sandbox/fortranpython/navaro/alloc.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import numpy as np 
import navaro as na

na.f90module.test_array()
#print len(na.f90module.farray) # na.f90module.farray = NoneType
na.f90module.init(10)
print(len(na.f90module.farray))
na.f90module.test_array()
